source/OrderedTypes/PredicativeLFP.lagda

```diff
@@ -0,0 +1,1704 @@
+   ↓ᴮf-equiv-S-tot a =
+    Σ-cong' (λ z → z ≤ᴮ f a)
+            ((λ z → (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds))
+            ↓ᴮf-equiv-S
+    where
+     ↓ᴮf-equiv-S : (z : B)
+                 → (z ≤ᴮ f a)
+                 ≃ (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds
+     ↓ᴮf-equiv-S z =
+      ⌜ prop-≃-≃-↔ fe ≤ᴮ-is-prop-valued ∥∥-is-prop ⌝⁻¹ (↓ᴮf-to-S , S-to-↓ᴮf)
+      where
+       ↓ᴮf-to-S : z ≤ᴮ f a
+                → (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds
+       ↓ᴮf-to-S o = ∣ (a , ⌜ ≃-Lift 𝓤 (z ≤ᴮ f a) ⌝ o , reflexivity-of L a) ∣
+       S-to-↓ᴮf : (Ǝ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds) holds
+                → z ≤ᴮ f a
+       S-to-↓ᴮf = ∥∥-rec ≤ᴮ-is-prop-valued S-to-↓ᴮf'
+        where
+         S-to-↓ᴮf' : Σ a' ꞉ ⟨ L ⟩ , (z , a') ∈ ϕ × (a' ≤ a) holds → z ≤ᴮ f a
+         S-to-↓ᴮf' (a' , o , r) =
+          ≤-to-≤ᴮ (transitivity-of L (β z) (f a') (f a)
+                                   (≤ᴮ-to-≤
+                                   (⌜ ≃-Lift 𝓤 (z ≤ᴮ f a') ⌝⁻¹ o))
+                                   (f-mono a' a r))
+   i : is-local ϕ
+   i a = (small-↓ᴮ (f a) , ↓ᴮf-equiv-S-tot a)
+   G : (x : ⟨ L ⟩) → (f x is-lub-of (S ϕ x , β ∘ S-to-base ϕ x)) holds 
+   G x = (fx-is-upbnd , fx-is-least-upbnd)
+    where
+     fx-is-upbnd : (f x is-an-upper-bound-of (S ϕ x , β ∘ S-to-base ϕ x)) holds
+     fx-is-upbnd (b , e) = S-to-fx-upbnd e
+      where
+       S-to-fx-upbnd : (Ǝ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ x) holds) holds
+                     → (β b ≤ f x) holds
+       S-to-fx-upbnd = ∥∥-rec (holds-is-prop (β b ≤ f x)) S-to-fx-upbnd'
+        where
+         S-to-fx-upbnd' : Σ a' ꞉ ⟨ L ⟩ , (b , a') ∈ ϕ × (a' ≤ x) holds
+                        → (β b ≤ f x) holds
+         S-to-fx-upbnd' (a' , o , r) =
+          transitivity-of L (β b) (f a') (f x)
+                          (≤ᴮ-to-≤ (⌜ ≃-Lift 𝓤 (b ≤ᴮ f a') ⌝⁻¹ o))
+                          (f-mono a' x r)
+     fx-is-least-upbnd : ((u , _) : upper-bound (S ϕ x , β ∘ S-to-base ϕ x))
+                       → (f x ≤ u) holds
+     fx-is-least-upbnd (u , is-upbnd) =
+      (is-least-upper-boundᴮ (f x))
+       (u , λ z → is-upbnd (⌜ ↓ᴮf-equiv-S-tot x ⌝ z))
+   H : (x : ⟨ L ⟩) → (Γ ϕ i) x ＝ f x
+   H x =
+    reindexing-along-equiv-＝-sup
+     L (id , id-is-equiv (S ϕ x)) (β ∘ S-to-base ϕ x)
+     ((Γ ϕ i) x) (f x) (sup-of-small-fam-is-lub  L (β ∘ S-to-base ϕ x) (i x))
+     (G x)
+
+ ind-def-from-mono-map : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                       → is-monotone-endomap L f
+                       → 𝓟 {𝓤 ⊔ 𝓥} (B × ⟨ L ⟩)
+ ind-def-from-mono-map f f-mono = pr₁ (mono-map-give-local-ind-def f f-mono)
+
+ local-from-mono-map : (f : ⟨ L ⟩ → ⟨ L ⟩)
+                     → (f-mono : is-monotone-endomap L f)
+                     → is-local (ind-def-from-mono-map f f-mono)
+ local-from-mono-map f f-mono =
+   pr₁ (pr₂ (mono-map-give-local-ind-def f f-mono))
+
+ f-＝-Γ-from-mono-map : (f : ⟨ L ⟩ → ⟨ L ⟩)
```

Comment:
Maybe `local-ind-def-is-section-of-Γ`? Also, the abbreviation `mono` is not great because it clashes with the terminology in category theory.

Reply:
Can you elaborate on why you see this as a section?

Reply:
I've fixed the mono abbreviation in type names btw and I agree this name could be better. It's just not clicking why it is a section as there are only two functions involved and we are saying they are equal.

Reply:
I was thinking it's like a section, because it (roughly) says that if you start with a monotone map `f` and pass its associated "inductive definition" to `Γ`, then you recover `f`, i.e. `Γ` after `ind-def` is the identity.

Reply:
Oh I do kinda see what you mean.